### DISCUSSION TITLE
VPN protection for Prowlarr

### DISCUSSION THREAD

**Discussion by defhead** [ahembree/ansible-hms-docker#42] [Q&A] [answered]
Currently in this stack, only Transmission is protected with a VPN and it's great. But in addition to the BitTorrent client, another service communicates with the trackers, Prowlarr. It indexes and downloads files from it.

And I was wondering..  What is your opinion about the fact that on that side, the trackers will still see our public IP? Isn't an issue and how could we protect it?

Thank you.

**Comment by ahembree** [comment, marked as answer]
I believe Prowlarr (and the other Starr apps) have support for proxies and the Transmission container comes with a proxy you can use to route traffic from those apps through the VPN (port 8888), but I've had issues with it working reliably since there's a known issue (or there was, haven't checked in a bit) where the VPN IP within the container changes and causes the proxy port to no longer be listening on the correct IP, which ends up "breaking" the proxy route.

Edit: Found the issue here: https://github.com/haugene/docker-transmission-openvpn/issues/2489. Looks like it has been resolved in this PR: https://github.com/haugene/docker-transmission-openvpn/pull/2678
